### ISSUE TITLE
Add support for VMware vSAN clusters

### ISSUE THREAD

**Issue by Redeux** [hashicorp/terraform-provider-vsphere#1091]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Add support for vSAN clusters

### Potential Terraform Configuration

```hcl
resource "vsphere_vnic" "v1" {
    host                    = data.vsphere_host.h1.id
    
    distributed_switch_port = vsphere_distributed_virtual_switch.d1.id
    distributed_port_group  = vsphere_distributed_port_group.p1.id
    
    ipv4 {
        dhcp = true
    }

    netstack = "vsan"
}

resource "vsphere_compute_cluster" "compute_cluster" {
  name            = "terraform-compute-cluster-test"
  datacenter_id   = "ata.vsphere_datacenter.dc.id
  host_system_ids = [data.vsphere_host.hosts.*.id]

  drs_enabled          = true
  drs_automation_level = "fullyAutomated"

  ha_enabled = true

  vsan_enabled               = true
  vsan_mode                  = "manual"
  vsan_health_check_interval = 1
}

resource "vsphere_vsan_disk_group" "vsan_group_h1" {
    host_system_ids = [data.vsphere_host.hosts.*.id]

    cache_tier    = "Hard Disk 1"
    capacity_tier = ["Hard Disk 2", "Hard Disk 3"]
}

resource "vsphere_vm_storage_policy" "vsan_storage_policy" {
    name      = "vsan_storage_policy"
    datastore = vsphere_compute_cluster.compute_cluster.vsan.datastore.id
    
    failure_tolerance        = 1
    disk_stripes_per_object  = 1
    read_cache_reservation   = 0
    object_space_reservation = 0
    force_provisioning       = false
}

resource "vsphere_virtual_machine" "vm1" {
    name           = "vm1"   
    storage_policy = vsphere_vm_storage_policy.vsan_storage_policy.id
}

```

### References

None

**Comment by duncanyb** [comment]
👍

Would like to see this fixed asap indeed!

**Comment by Krayg** [comment]
Does it really need "vsan_mode                  = "manual"" - isn't this the default/only option now?

**Comment by lamw** [comment]
+1 

@Krayg Agreed, automatic mode isn't supported anymore and there's only single default value.

**Comment by AdrianBegg** [comment]
+1

Enhancement to the vsphere_vm_storage_policy to support the vSAN rulesets would be for us a priority for us for templated deployments of VMC SDDCs.

**Comment by bill-rich** [comment]
Basic vSAN support has been added in #1151, and released in v1.22.0,
